From f420d534b143fd4d72a7d6be21baf111536fc633 Mon Sep 17 00:00:00 2001
From: Richard Nelson <wabz@pidgin.im>
Date: Sat, 29 Jul 2006 09:39:43 +0000
Subject: [PATCH] [gaim-migrate @ 16594] Enhancement patch from wabz (Richard
 Nelson) to show the title of the conversation, and to show
 the alias of the buddy in the conversation if set.

committer: Sadrul Habib Chowdhury <sadrul@pidgin.im>

--HG--
extra : convert_revision : faa041313ba12b20dc3aced3c51f81a343fba0c4
---
 console/gntconv.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/console/gntconv.c b/console/gntconv.c
index 9895365..c22d6fd 100644
--- a/console/gntconv.c
+++ b/console/gntconv.c
@@ -162,7 +162,7 @@ gg_create_conversation(GaimConversation *conv)
 	conv->ui_data = ggc;
 
 	type = gaim_conversation_get_type(conv);
-	title = g_strdup_printf(_("%s"), gaim_conversation_get_name(conv));
+	title = g_strdup_printf(_("%s"), gaim_conversation_get_title(conv));
 	ggc->window = gnt_box_new(FALSE, TRUE);
 	gnt_box_set_title(GNT_BOX(ggc->window), title);
 	gnt_box_set_toplevel(GNT_BOX(ggc->window), TRUE);
@@ -244,9 +244,9 @@ static void
 gg_write_im(GaimConversation *conv, const char *who, const char *message,
 		GaimMessageFlags flags, time_t mtime)
 {
+	GaimAccount *account = gaim_conversation_get_account(conv);
 	if (flags & GAIM_MESSAGE_SEND)
 	{
-		GaimAccount *account = gaim_conversation_get_account(conv);
 		who = gaim_connection_get_display_name(gaim_account_get_connection(account));
 		if (!who)
 			who = gaim_account_get_alias(account);
@@ -254,7 +254,13 @@ gg_write_im(GaimConversation *conv, const char *who, const char *message,
 			who = gaim_account_get_username(account);
 	}
 	else if (flags & GAIM_MESSAGE_RECV)
+	{
+		GaimBuddy *buddy;
 		who = gaim_conversation_get_name(conv);
+		buddy = gaim_find_buddy(account, who);
+		if (buddy)
+			who = gaim_buddy_get_contact_alias(buddy);
+	}
 
 	gg_write_common(conv, who, message, flags, mtime);
 }
-- 
1.7.10.4

